src/main/java/net/hypixel/nerdbot/generator/MinecraftImage.java

```diff
@@ -171,19 +143,48 @@ private Graphics2D initG2D(int width, int height) {
      * @return The initialized font.
      */
     @Nullable
-    private Font initFont(String path, float size) {
+    private static Font initFont(String path, float size) {
         Font font;
         try {
             InputStream fontStream = ItemGenCommand.class.getResourceAsStream(path);
             if (fontStream == null) {
                 return null;
             }
             font = Font.createFont(Font.TRUETYPE_FONT, fontStream).deriveFont(size);
-            ge.registerFont(font);
         } catch (IOException | FontFormatException e) {
             e.printStackTrace();
             return null;
         }
         return font;
     }
+
+    /**
+     * Initializes all required fonts before executing
+     *
+     * @return If registering fonts was successful
+     */
+    public static boolean registerFonts() {
+        if (fontsRegistered)
+            return true;
+
+        if (minecraftFonts[0] == null) {
+            minecraftFonts[0] = initFont("/Minecraft/minecraft.ttf", 16f);
+        }
+        if (minecraftFonts[1] == null) {
+            minecraftFonts[1] = initFont("/Minecraft/3_Minecraft-Bold.otf", 22f);
+        }
+        if (minecraftFonts[2] == null) {
+            minecraftFonts[2] = initFont("/Minecraft/2_Minecraft-Italic.otf", 22f);
+        }
+        if (minecraftFonts[3] == null) {
+            minecraftFonts[3] = initFont("/Minecraft/4_Minecraft-BoldItalic.otf", 22f);
+        }
+        if (sansSerif == null) {
+            sansSerif = new Font("SansSerif", Font.PLAIN, 20);
+        }
+
+        fontsRegistered = true;
```

Comment:
initFont is nullable, so these should be checked before setting fontsRegistered to true